### DISCUSSION TITLE
DeepONet without labeled data

### DISCUSSION THREAD

**Discussion by HamidTGanji** [lululxvi/deepxde#1270] [Q&A]
Hi LuLu.
Thank you for providing this great package.
There are good number of examples in your documentation which show how to train the PINN based on governing differential equation and without having labeled/observed data. However, I can't find the same issue for DeepONet topic. Is there any example for DeepONet that I can use it as a guide for this task? 

Thanks,
Hamid

**Comment by pescap** [comment]
Hi Hamid, 
As far as I know, physics can be incorporated to the DeepONet loss and produce PI-DeepONets. See for example [this repository](https://github.com/PredictiveIntelligenceLab/Long-time-Integration-PI-DeepONets
).

**Reply by HamidTGanji** [reply]
Thank you for your reply pescap.
Yes, I also read some papers which used the PI-DeepONet. However, I wanted to see an example of that in the DeepXDE package.
My guess is that I should probably use dde.data.PDEOperator or dde.data.PDEOperatorCartesianProd. But I don't know how exactly I can do that.

**Reply by pescap** [reply]
@lululxvi would that require to change the source code? It would be interesting to add a PI-DeepONet example and/or a more complex DeepONet example into documentation. Any suggestion?

**Reply by pescap** [reply]
FYI: https://github.com/lululxvi/deepxde/issues/679

**Reply by HamidTGanji** [reply]
I appreciate your help, pescap.
However, I'm a bit confused. As we see in answer #679, we need a dataset for training the DeepONet. To me, dataset means labeled data or observations. In PI-DeepONet, we don't need any dataset for training purposes. I can't relate the two sentences of that solution to each other.

**Reply by lululxvi** [reply]
See https://github.com/lululxvi/deepxde/pull/1311
